File: samplejava/src/main/java/pyxis/uzuki/live/richutils/samplejava/PermissionActivity.java
package pyxis.uzuki.live.richutils.samplejava;

import android.Manifest;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.widget.Button;

import butterknife.BindView;
import pyxis.uzuki.live.richutilskt.utils.RPermission;
import pyxis.uzuki.live.richutilskt.utils.RichUtils;

/**
 * Created by pyxis on 2017. 6. 12..
 */

public class PermissionActivity extends BaseActivity {
    @BindView(R.id.btnCall) Button btnCall;
    @BindView(R.id.btnLocation) Button btnLocation;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_permission);

        btnCall.setOnClickListener(view -> {
            String[] arrays = new String[]{Manifest.permission.CALL_PHONE};

            RPermission.instance.checkPermission(this, arrays, (integer, strings) -> {
                if (integer == RPermission.PERMISSION_GRANTED) {
                    // 성공
                } else {
                    // 실패
                }
                RichUtils.alert(PermissionActivity.this,
                        "Permission result ->" + integer + " / Requested Permission: " + TextUtils.join(",", strings));
            });
        });

        btnLocation.setOnClickListener(view -> {
            String[] arrays = new String[]{Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION};

            RPermission.instance.checkPermission(this, arrays, (integer, strings) -> {
                if (integer == RPermission.PERMISSION_GRANTED) {
                    // 성공
                } else {
                    // 실패
                }
                RichUtils.alert(PermissionActivity.this,
                        "Permission result ->" + integer + " / Requested Permission: " + TextUtils.join(",", strings));
            });
        });
    }
}

File: sample/src/main/java/pyxis/uzuki/live/richutilssample/PermissionActivity.kt
package pyxis.uzuki.live.richutilssample

import android.Manifest
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import kotlinx.android.synthetic.main.activity_permission.*
import pyxis.uzuki.live.richutilskt.utils.RPermission
import pyxis.uzuki.live.richutilskt.utils.alert
import pyxis.uzuki.live.richutilskt.utils.getIMEI

class PermissionActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_permission)

        btnCall.setOnClickListener {
            val arrays: Array<String> = arrayOf(Manifest.permission.CALL_PHONE)
            RPermission.instance.checkPermission(this, array = arrays, callback = { resultCode: Int, list: List<String> ->
                resultDisplay(resultCode, list)
            })
        }

        btnSMS.setOnClickListener {
            val arrays: Array<String> = arrayOf(Manifest.permission.SEND_SMS)
            RPermission.instance.checkPermission(this, array = arrays, callback = { resultCode: Int, list: List<String> ->
                resultDisplay(resultCode, list)
            })
        }

        btnStorage.setOnClickListener {
            val arrays: Array<String> = arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE)
            RPermission.instance.checkPermission(this, array = arrays, callback = { resultCode: Int, list: List<String> ->
                resultDisplay(resultCode, list)
            })
        }

        btnLocation.setOnClickListener {
            val arrays: Array<String> = arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION)
            RPermission.instance.checkPermission(this, array = arrays, callback = { resultCode: Int, list: List<String> ->
                resultDisplay(resultCode, list)
            })
        }

        btnState.setOnClickListener {
            val arrays: Array<String> = arrayOf(Manifest.permission.READ_PHONE_STATE)
            RPermission.instance.checkPermission(this, array = arrays, callback = { resultCode: Int, list: List<String> ->
                resultDisplay(resultCode, list)
            })
        }
    }

    fun resultDisplay(resultCode: Int, list: List<String>) {
        alert(message = "Permission result -> $resultCode / Requested Permission: ${TextUtils.join(",", list)}")

        if (list.contains(Manifest.permission.READ_PHONE_STATE)) {
            alert(message = "Your device IMEI -> ${getIMEI()}")
        }
    }
}
